### ISSUE TITLE
velbus plugin install error

### ISSUE THREAD

**Issue by maxs99** [domogik/domogik#285]
Hi have i think so a running install but when i try to install the velbus plugin these ones comes up

root@ha:~# dmg_package --install http://github.com/Cereal2nd/domogik-plugin-velbus/archive/1.2.zip
Start downloading http://github.com/Cereal2nd/domogik-plugin-velbus/archive/1.2.zip
Error while downloading the package : Traceback (most recent call last):
  File "/opt/dmg/domogik-0.4.2/src/domogik/bin/package.py", line 375, in download_from_url
    mime = magic.from_buffer(peek, mime=True)
AttributeError: 'module' object has no attribute 'from_buffer'

Traceback (most recent call last):
  File "/usr/local/bin/dmg_package", line 9, in <module>
    load_entry_point('Domogik==0.4.2', 'console_scripts', 'dmg_package')()
  File "/opt/dmg/domogik-0.4.2/src/domogik/bin/package.py", line 605, in main
    pkg = PackageInstaller()
  File "/opt/dmg/domogik-0.4.2/src/domogik/bin/package.py", line 128, in __init__
    self.install(self.options.install)
  File "/opt/dmg/domogik-0.4.2/src/domogik/bin/package.py", line 169, in install
    downloaded_file, mime = self.download_from_url(package)
  File "/opt/dmg/domogik-0.4.2/src/domogik/bin/package.py", line 375, in download_from_url
    mime = magic.from_buffer(peek, mime=True)
AttributeError: 'module' object has no attribute 'from_buffer'
root@ha:~#

thx max

**Comment by tikismoke** [comment]
Seems it is the download that make problem.
To be sure can you try downloading the zip and restart the installation with the local file?

**Comment by maxs99** [comment]
yes I can download the file via wget command 
but the local install i did not now howto

max

ps. every addons makes the same error

**Comment by tikismoke** [comment]
Just try a dmg_package --install /any/path/where/you/just/download/the/zip/file/1.2.zip

**Comment by tikismoke** [comment]
I you need more help go to irc #domogik for real chat help ;)

**Comment by maxs99** [comment]
arg mph copy past errror
manual download and lokal install works

**Comment by maxs99** [comment]
sorry guys a new error
i get an 500 - Internal server error after 3 velbus devices
a reinstall of the plugin did not solve the problem
the 500 - Internal server error is also on create new scenario

max

ps my version is a 0.4.2

**Comment by Cereal2nd** [comment]
can you get me the output of admin.py-f at the moment the 500 happens?

This is probably a generic bug and has nothing todo with the velbus code

**Comment by maxs99** [comment]
$ python admin.py -f
plugin_weather
plugin_velbus
brain_base
is manager
2016-03-30 07:07:19,599 domogik-admin INFO Looking for launched instances of 'ad                                                  min'
2016-03-30 07:07:19,615 domogik-admin INFO Process found : domogik   1844  0.0                                                    2.4  36192 26196 ?        S    Mar29   0:00 /usr/bin/python /opt/dmg/domogik-0.4                                                  .2/src/domogik/bin/admin.py
2016-03-30 07:07:19,615 domogik-admin INFO Process found : domogik   1847  0.1                                                    3.3 446332 35340 ?        Sl   Mar29   1:15 /usr/bin/python /opt/dmg/domogik-0.4                                                  .2/src/domogik/bin/admin.py
2016-03-30 07:07:19,615 domogik-admin INFO There are already existing processes.
2016-03-30 07:07:19,616 domogik-admin DEBUG __del__ Single client
2016-03-30 07:07:19,616 domogik-admin DEBUG the stack is :
2016-03-30 07:07:19,671 domogik-admin DEBUG     (<frame object at 0xb5e4adec>, '                                                  /opt/dmg/domogik-0.4.2/src/domogik/common/plugin.py', 883, '__del__', ['                                                              for elt in inspect.stack():\n'], 0)
2016-03-30 07:07:19,672 domogik-admin DEBUG force_leave called
2016-03-30 07:07:19,672 domogik-admin DEBUG the stack is :
2016-03-30 07:07:19,672 domogik-admin DEBUG     (<frame object at 0xb5f793fc>, '                                                  /opt/dmg/domogik-0.4.2/src/domogik/common/plugin.py', 897, 'force_leave', ['                                                              for elt in inspect.stack():\n'], 0)
2016-03-30 07:07:19,672 domogik-admin DEBUG     (<frame object at 0xb5e4adec>, '                                                  /opt/dmg/domogik-0.4.2/src/domogik/common/plugin.py', 886, '__del__', ['                                                              self.force_leave()\n'], 0)
2016-03-30 07:07:19,695 domogik-admin DEBUG __del__ Single client
2016-03-30 07:07:19,696 domogik-admin DEBUG the stack is :
2016-03-30 07:07:19,696 domogik-admin DEBUG     (<frame object at 0xb5e51b2c>, '                                                  /opt/dmg/domogik-0.4.2/src/domogik/common/plugin.py', 883, '__del__', ['                                                              for elt in inspect.stack():\n'], 0)
2016-03-30 07:07:19,696 domogik-admin DEBUG force_leave called
2016-03-30 07:07:19,696 domogik-admin DEBUG the stack is :
2016-03-30 07:07:19,697 domogik-admin DEBUG     (<frame object at 0xb5f793fc>, '                                                  /opt/dmg/domogik-0.4.2/src/domogik/common/plugin.py', 897, 'force_leave', ['                                                              for elt in inspect.stack():\n'], 0)
2016-03-30 07:07:19,697 domogik-admin DEBUG     (<frame object at 0xb5e51b2c>, '                                                  /opt/dmg/domogik-0.4.2/src/domogik/common/plugin.py', 886, '__del__', ['                                                              self.force_leave()\n'], 0)
$

**Comment by maxs99** [comment]
I have played around with the database and found out when i delete the velbus devices the 500 error are gone

but i am not able to switch any light from the domoweb but it is possible from the admin interface
I get also an error from the scenarios "unexpected result from scenario manager"
and the shutter has no xpl settings

so i delet everything and did a reinstall
but the errors are the same

**Comment by maxs99** [comment]
i found the problem !!!
it comes from the velbus plugin when i add the dimmer the 500 error comes up

**Comment by maxs99** [comment]
so a last test 
i have deleted the dimmer and put at 3 position an other velbus device no problem
but when i insert the dimmer the 500 error comes up

one thing i found out 
the plugin did not read out any values from the devices so domoweb did not showing anything

**Comment by Cereal2nd** [comment]
i have no dimmers at home so i can not test the velbus dimmers.

Will need to buy at least one to make sure i can really test this ...

**Comment by maxs99** [comment]
> i have no dimmers at home so i can not test the velbus dimmers. Will need to buy at least one to make sure i can really test this ...

can i help with my dimmer??

> one thing i found out
the plugin did not read out any values from the devices so domoweb did not showing anything

any idea why the plugin do not read the state from the modules ???

**Comment by maxs99** [comment]
Hey Guys come no one an idea??????

**Comment by jschaeke** [comment]
I am waiting on the release 0.5 (end of april?) to install domogik for my home automation. I have 11 velbus dimmers, so I will certainly report here if I face similar issues.

**Comment by tikismoke** [comment]
Here they're is 2 different problems.

The installation part and the dimmer requests.
Let's fo cus for installation here and move your dimmer request to the corresponding plugin.
Search domogik-plugin on github to find the good one.

**Comment by fritz-smh** [comment]
I agree with tikismoke, there are 2 issues, one for velbus. For this one, please create a ticket here : https://github.com/Cereal2nd/domogik-plugin-velbus/pulls

The other one is about packages installation.
About this second one : 

1. do you still have an issue ?
2. if so, do you have this issue with an online package (dmg_package -i http://.....) or a local package (zip downloaded, then dmg_package -i /path/to/zip) or both ?

Thanks

**Comment by maxs99** [comment]
Okay,
yes my issues still exits,

only a local install of the plugin is possible 
the install with an online package ends with the error in my first post

the second error was a wihle unclear for me, now i will will open a ticket in the right section

Max

**Comment by maxs99** [comment]
@Cereal2nd 
you are also reading these issue and i am not able to open a ticket on your github so please comment something here

Max

**Comment by Cereal2nd** [comment]
yeah i'm reading, i need to find some time to check out the reason why we
get the errors in the admin

**Comment by Cereal2nd** [comment]
i think i solved the admin error

update to the latest verdion of develop for domogik AND plugin-velbus

**Comment by maxs99** [comment]
@Cereal2nd  you have found and fixed the problem faster as i could test it

**Comment by Cereal2nd** [comment]
It was just a typo in the json file

**Comment by jschaeke** [comment]
I just installed on the rpi3 the latest develop of both domogik as this plugin. The installation went smooth but unfortunately the dimmer won't work. Here a gist of what happens if I click on the physical switch (VMB6PBN) to control a dimmer: https://gist.github.com/anonymous/e0c56ad5c3b04b136739a162b66ea8eb

If it helps I am willing to open up an ssh to my rpi..

**Comment by Cereal2nd** [comment]
was this a dimming to 0?

2016-04-16 15:17:59,462 domogik-velbus INFO Sending MQ status: dev:115 chan:1 dt:DT_Scaling sen:2 value:0

**Comment by jschaeke** [comment]
So here again a log: 
I turned on the light on dimmer 115 using a switch VMB6PBN that has a led (indicator) at 24:01, the light turns on (gradually in 2 seconds) 24:03.
at 24:11 I turned of the light with same switchand light fades and is off at 24:13.
https://gist.github.com/jschaeke/7715ccee11f73fc4008fc999f94d0db6

**Comment by Cereal2nd** [comment]
what would help is if you could get me the following info:

2016-04-16 21:24:01,440 domogik-velbus DEBUG starting parser: 0ffb7308b8015000000000007204
2016-04-16 21:24:01,440 domogik-velbus DEBUG Received message with type: 'dimmer channel status' address: 115 module: VMBDMI(21)
2016-04-16 21:24:01,440 domogik-velbus DEBUG Messagetype module specifick parser not implemented
2016-04-16 21:24:01,441 domogik-velbus INFO Sending MQ status: dev:115 chan:1 dt:DT_Scaling sen:2 value:0

for setting the dimmer to off, on and some values in between (let me know the value)
This way i can simulate a dimmer here at home.

**Comment by jschaeke** [comment]
And is there any log/method you were thinking of I could provide? I already tried to sniff the serial port but that didn't give anything useful.

**Comment by Cereal2nd** [comment]
it is all in the velbus.log

so control the dimmer via velbuslink and then the info should be in the log

**Comment by jschaeke** [comment]
Ok it took me some time to install jvelbus (velserv didnt work). So now that's done I've turned on the dimmer on and off straight from Velbuslink: I commented the 2 lines responsible for turning on and off the dimmer (I resent them to confirm). The value seems to be 7. As I see 'set dim value (07)' in the commando log. 
https://gist.github.com/jschaeke/b7fdcc5562718e75f2246d2ca7824786

**Comment by Cereal2nd** [comment]
ok the problem must be in this part of the code

`    def _process_184(self, data):
        """
           Process a 184 Message
           Dimmer channel status => send out when the dimmer status is changed
        """
        for channel in self._byte_to_channels(data[5]):
            level = -1
            level = (100 / 255 ) * ord(data[7])
            if level != -1:
                self._callback( str(ord(data[2])), str(channel), 'DT_Scaling', level )
`

can you add some debug here?
print the data array and print the level, probably something is happening here

**Comment by jschaeke** [comment]
Ok I will try this evening. Oh since I have yesterday jvelvusd running, I could open it up to the internet for you to connect (I don't mind for a couple of days).

**Comment by maxs99** [comment]
Sorry for my late answer,

but after a upgrade to the developer version the admin won´t start,
so i will give up and do it on fresh install.

@jschaeke
witch problen did you have with velservc?
mine runs over months without any problem, and is even forwarded to the 
inet.
and i could connect velbuslink to my external ip without any probs

Am 18.04.2016 um 11:36 schrieb jschaeke:

**Comment by jschaeke** [comment]
@maxs99 With velserv I saw velbuslink getting a connection, but no commands were transmitted/received over it. I saw people with similar issues and they had switched to jvelbusd. It was a pain to install it because of it dependency to a serial library... 

I just installed the docker image successfully on my mac os. I will use this one from now and have just the jvelbusd running on the pi as long as I'm in this debugging-mode ;-)

**Comment by maxs99** [comment]
@jschaeke
my install runs on Proxmox in a Container and a Raspi with velserv makes 
the usb to ethernet connection.
Why i use velserv i think its possible to interconnect two velbus 
installations, at jvelbusd i looked often at it but i did not now how to 
install on my raspi.

Am 18.04.2016 um 11:53 schrieb jschaeke:

**Comment by jschaeke** [comment]
Hmm problems here. Since yesterday I can't control my regular relay anymore which worked before? Both serial as socket method fail when running domogik straight on the rpi. A reinstall of domogik + clean database didn't help.
So today I started from a fresh docker domogik image on my mac and connect straight to the jvelbusd  which is the sole thing running on the rpi. The jvelbusd is ok as I can successfully connect and control the devices with Velbuslink. However, I get always the same error now on my docker-image:

2016-04-18 19:26:09,877 domogik-velbus DEBUG MQ Request received : <MQMessage(action=client.cmd, data='{u'device_id': 1, u'command_id': 1, u'level': 255}')>
2016-04-18 19:26:09,877 domogik-velbus DEBUG received set_level for 103
2016-04-18 19:26:09,877 domogik-velbus ERROR Request to set a level on a device, but the device is not known. address 103
2016-04-18 19:26:09,877 domogik-velbus ERROR Can not Send MQ status, sensor not found
2016-04-18 19:26:09,878 domogik-velbus DEBUG device: 103 channel: 3 dt: ['DT_ScalingDT_Switch']
2016-04-18 19:26:10,648 domogik-velbus DEBUG MQ Request received : <MQMessage(action=client.cmd, data='{u'device_id': 1, u'command_id': 1, u'level': 0}')>
2016-04-18 19:26:10,649 domogik-velbus DEBUG received set_level for 103
2016-04-18 19:26:10,649 domogik-velbus ERROR Request to set a level on a device, but the device is not known. address 103
2016-04-18 19:26:10,649 domogik-velbus ERROR Can not Send MQ status, sensor not found
2016-04-18 19:26:10,649 domogik-velbus DEBUG device: 103 channel: 3 dt: ['DT_ScalingDT_Switch']
2016-04-18 19:26:21,213 domogik-velbus DEBUG Send client status : alive

So I have no working domogik anymore... I checked the latest commits, but there's nothing in there last 2 days that broke it.  I think I'm gonna start again from scratch (new debian install).  I hope I can supply more debug soon

**Comment by Cereal2nd** [comment]
we need to split this in 2 parts

1- receiving the status => this is what i'm trying to fix now
2- sending the status => can be fixed once 1 is ok

for now it seems we always report level=0 wich is not correct, so the parser is making a mistake somewhere not sure where at the moment.

is it correct that this is device 115 channel 1?

**Comment by Cereal2nd** [comment]
i just pushed the fix (i think) can you retest?

just update domogik-plugin-velbus repository and restart the plugin

**Comment by jschaeke** [comment]
Yes, it is a 1 channel dimmer module (so channel 1). And I will test asap (probably this evening).

**Comment by jschaeke** [comment]
Unfortunately, the latest code crashed the velbus plugin completely (can't get it started):
2016-04-20 21:12:12,766 domogik-velbus DEBUG Send client status : alive
2016-04-20 21:12:20,263 domogik-velbus DEBUG Delete the file /var/run/domogik//velbus_return_code_2616
2016-04-20 21:12:20,265 domogik-velbus DEBUG __del__ Single client
2016-04-20 21:12:20,266 domogik-velbus DEBUG force_leave called

I had to revert to checkout 796290be5a2fa948ddb5492c7754d69737c6c076 to get it back working.

**Comment by jschaeke** [comment]
I updated the code to:
` for channel in self._byte_to_channels(data[5]):
            level = -1
            level = 2.5 * ord(data[7])
            self._log.debug("*****184 process data : {0} , channel : {1} ,  level : {2}".format('|'.join(map(str,map(ord,data))),str(channel), level))
            self._log.debug("*****data 7 :" + str(ord(data[7])))
            if level != -1:
                self._callback( str(ord(data[2])), str(channel), 'DT_Scaling', level )
`
but the slider won't move in domoweb and last level sensor value in admin screen remains 0%:

`2016-04-20 21:43:08,387 domogik-velbus DEBUG Received message with type: 'dimmer channel status' address: 115 module: VMBDMI(21)
2016-04-20 21:43:08,388 domogik-velbus DEBUG Messagetype module specifick parser not implemented
2016-04-20 21:43:08,389 domogik-velbus DEBUG *****184 process data : 15|251|115|8|184|1|80|20|128|0|0|0|222|4 , channel : 1 ,  level : 50.0
2016-04-20 21:43:08,389 domogik-velbus DEBUG *****data 7 :20`

**Comment by Cereal2nd** [comment]
ticket moved to
https://github.com/Cereal2nd/domogik-plugin-velbus/issues/2

as this is a pure velbus problem
